homeassistant/components/mill/climate.py

```diff
@@ -84,6 +90,21 @@ async def set_room_temp(service: ServiceCall) -> None:
         DOMAIN, SERVICE_SET_ROOM_TEMP, set_room_temp, schema=SET_ROOM_TEMP_SCHEMA
     )
 
+    async def max_heating_power(entity: MillHeater, service: ServiceCall) -> None:
+        """Limit heating power."""
+        await mill_data_coordinator.mill_data_connection.max_heating_power(
+            entity.heater_id, service.data[ATTR_MAX_HEATING_POWER]
+        )
+
+    async_get_current_platform().async_register_entity_service(
+        SERVICE_MAX_HEATING_POWER,
+        schema={
+            vol.Required(ATTR_ENTITY_ID): cv.entity_id,
+            vol.Required(ATTR_MAX_HEATING_POWER): vol.Range(min=0, max=2000),
+        },
+        func=max_heating_power,
+    )
```

Comment:
Why don't we make this a number entity?

Reply:
Good suggestion.
I will make another PR for that.